crates/shared/src/order_validation.rs

```diff
@@ -928,8 +931,22 @@ pub struct Amounts {
 ///
 /// Note that this check only looks at the order's limit price and the market
 /// price and is independent of amounts or trade direction.
-pub fn is_order_outside_market_price(order: &Amounts, quote: &Amounts) -> bool {
-    (order.sell + order.fee).full_mul(quote.buy) < (quote.sell + quote.fee).full_mul(order.buy)
+pub fn is_order_outside_market_price(
+    order: &Amounts,
+    quote: &Amounts,
+    kind: model::order::OrderKind,
+) -> bool {
+    match kind {
+        OrderKind::Buy => {
+            order.sell.full_mul(quote.buy) < (quote.sell + quote.fee).full_mul(order.buy)
```

Comment:
Buy order stayed the same as before, as we do need to add fee to sell amount in this case